<launch>

    <!--NAMES ARE GIVEN IN THE GROUP TAG, ATTRIBUTE NS-->
    <group ns="cart">
        <param name="program_location" type="string" value="nodes/programs/"/>
        <param name="object_module_name" type="string" value="cart"/>
        <param name="object_class_name" type="string" value="cart"/>

        <param name="frame_1_parent" type="string" value="world"/>
        <param name="frame_1" type="string" value="cart_frame"/>
        <param name="frame_1_updater" type="string" value="getConfigurationMatrixCart"/>
    </group>
    <group ns="arm">
        <param name="program_location" type="string" value="nodes/programs/"/>
        <param name="object_module_name" type="string" value="arm"/>
        <param name="object_class_name" type="string" value="arm"/>

        <param name="frame_1_parent" type="string" value="cart_frame"/>
        <param name="frame_1" type="string" value="frame_turntable"/>
        <param name="frame_1_updater" type="string" value="getConfigurationMatrixTurntable"/>

        <param name="frame_2_parent" type="string" value="frame_turntable"/>
        <param name="frame_2" type="string" value="frame_cantilever"/>
        <param name="frame_2_updater" type="string" value="getConfigurationMatrixCantilever"/>

        <param name="frame_3_parent" type="string" value="frame_cantilever"/>
        <param name="frame_3" type="string" value="frame_anchorpoint"/>
        <param name="frame_3_updater" type="string" value="getConfigurationMatrixAnchorPoint"/>

        <param name="frame_4_parent" type="string" value="frame_anchorpoint"/>
        <param name="frame_4" type="string" value="frame_gripper"/>
        <param name="frame_4_updater" type="string" value="getConfigurationMatrixGripper"/>
    </group>
    <group ns="carrier">
        <param name="program_location" type="string" value="nodes/programs/"/>
        <param name="object_module_name" type="string" value="carrier"/>
        <param name="object_class_name" type="string" value="carrier"/>

        <param name="frame_1_parent" type="string" value="world"/>
        <param name="frame_1" type="string" value="carrier_frame"/>
        <param name="frame_1_updater" type="string" value="getConfigurationMatrixCart"/>
    </group>


</launch>
